src/parameters.rs

```diff
@@ -61,13 +61,53 @@ impl From<Log> for ResultLog {
     }
 }
 
+/// The status of a transaction.
+#[derive(Debug, BorshSerialize, BorshDeserialize, PartialEq, Eq)]
+pub enum TransactionStatus {
+    Succeed(Vec<u8>),
+    Revert(Vec<u8>),
+    OutOfGas,
+    OutOfFund,
+    OutOfOffset,
+    CallTooDeep,
+}
+
+impl TransactionStatus {
+    pub fn is_ok(&self) -> bool {
+        matches!(*self, TransactionStatus::Succeed(_))
+    }
+
+    pub fn is_revert(&self) -> bool {
+        matches!(*self, TransactionStatus::Revert(_))
+    }
+
+    pub fn is_fail(&self) -> bool {
+        *self == TransactionStatus::OutOfGas
+            || *self == TransactionStatus::OutOfFund
+            || *self == TransactionStatus::OutOfOffset
+            || *self == TransactionStatus::CallTooDeep
+    }
+}
+
+impl AsRef<[u8]> for TransactionStatus {
+    fn as_ref(&self) -> &[u8] {
+        match self {
+            Self::Succeed(_) => b"SUCCESS",
+            Self::Revert(_) => b"ERR_REVERT",
+            Self::OutOfFund => b"ERR_OUT_OF_FUND",
```

Comment:
As I mentioned in the other PR, `OutOfFund` is not correct, but if you want to keep it for consistency with SputnikVM, we should at least use the correct description in the error itself:

```suggestion
            Self::OutOfFund => b"ERR_OUT_OF_FUNDS",
```